File: app-miyou/src/main/java/com/miyou/app/application/mission/usecase/MissionCompletionUseCase.java
package com.miyou.app.application.mission.usecase;

import com.miyou.app.domain.dialogue.model.UserId;
import com.miyou.app.domain.mission.model.MissionId;
import com.miyou.app.domain.mission.model.UserMission;
import reactor.core.publisher.Mono;

public interface MissionCompletionUseCase {

	Mono<UserMission> completeMission(UserId userId, MissionId missionId);
}

File: app-miyou/src/main/kotlin/com/miyou/app/application/mission/usecase/MissionCompletionUseCase.kt
package com.miyou.app.application.mission.usecase

import com.miyou.app.domain.mission.model.MissionId
import com.miyou.app.domain.mission.model.UserMission
import reactor.core.publisher.Mono

interface MissionCompletionUseCase {
    fun completeMission(
        userId: String,
        missionId: MissionId,
    ): Mono<UserMission>
}
